### PR TITLE
Add source gen implementation

### PR DESCRIPTION
**Pull request by NipunaMarcus**
### Update
Source generation for the formatting language feature is now implemented using a source generation which is implemented using java. 

### User Guidelines
To generate the SourceGen.java please follow the below guidelines.
- Open command line
- Navigate to *ballerina/composer/modules/web* 
- enter the command 
```
npm run gen-source-gen-java
```
you will be able to find the generated SourceGen.java file at *ballerina/language-server/modules/langserver-core/org.ballerinalang.langserver/SourceGen.java*

### CODE REVIEW

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/BallerinaTextDocumentService.java [hunk 1]
```diff
@@ -290,17 +301,36 @@ public BallerinaTextDocumentService(BallerinaLanguageServer ballerinaLanguageSer
...
-            // TODO: For the moment we replace the content with the ast. Need to implement the AST source-gen
-            String newDummyContent = ast.toString();
+            ClassLoader classLoader = BallerinaTextDocumentService.class.getClassLoader();
+            InputStream inputStream = classLoader.getResourceAsStream(SOURCE_GEN_FILE);
+            try {
+                Reader fileReader = new InputStreamReader(inputStream, "UTF-8");
+                engine.eval(fileReader);
+                fileReader.close();
+                Invocable invocable = (Invocable) engine;
+                newDummyContent = (String) invocable.invokeFunction("gen", ast.toString());
```
Comment (roonyh) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/BallerinaTextDocumentService.java]: Should this content still be called dummy ? It’s proper content now right?

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/BallerinaTextDocumentService.java [hunk 2]
```diff
@@ -290,17 +301,36 @@ public BallerinaTextDocumentService(BallerinaLanguageServer ballerinaLanguageSer
     @Override
     public CompletableFuture<List<? extends TextEdit>> formatting(DocumentFormattingParams params) {
         return CompletableFuture.supplyAsync(() -> {
+            ScriptEngine engine = new ScriptEngineManager().getEngineByName(JAVASCRIPT_ENGINE);
             TextDocumentServiceContext formatContext = new TextDocumentServiceContext();
             formatContext.put(DocumentServiceKeys.FILE_URI_KEY, params.getTextDocument().getUri());
             String fileContent = documentManager.getFileContent(CommonUtil.getPath(params.getTextDocument().getUri()));
+            String newDummyContent = fileContent;
```
Comment (nadeeshaan) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/BallerinaTextDocumentService.java]: Let's change the variable name to something else such as textEditContent

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/BallerinaTextDocumentService.java [hunk 3]
```diff
@@ -290,17 +301,36 @@ public BallerinaTextDocumentService(BallerinaLanguageServer ballerinaLanguageSer
...
+            } finally {
+                if (inputStream != null) {
+                    try {
+                        inputStream.close();
+                    } catch (IOException e) {
+                        // Ignore
+                    }
+                }
+            }
             TextEdit textEdit = new TextEdit(range, newDummyContent);
```
[the remark this answers is not in the document]
Comment (NipunaMarcus) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/BallerinaTextDocumentService.java]: Have to check whether it removes the content from the document...

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/BallerinaTextDocumentService.java (hunk 1 above)
Comment (NipunaMarcus) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/BallerinaTextDocumentService.java]: No dummy it should not 😀 will change it.

**Pull request merged**